mlflow/store/artifact/artifact_repo.py

```diff
@@ -144,6 +159,9 @@ def _download_file(self, remote_file_path, local_path):
         :param remote_file_path: Source path to the remote file, relative to the root
                                  directory of the artifact repository.
         :param local_path: The path to which to save the downloaded file.
+        :return: Either `None` (for synchronous download implementations) or an instance of
```

Comment:
Returning 2 types of objects depending on the implementation is confusing. I suggest not to touch this API. Instead, add a new non-abstract method, which can be used in future for other ArtifactRepo subclasses. Below code might not work as is, please pardon the lack of testing.

def _download_files(self, remote_file_paths, local_path):
    exceptions = []
    for f in concurrent.futures.as_completed([self.thread_pool.submit(
            self._download_file, path, local_path) for path in remote_file_paths]):
        try:
          f.result()
        except Exception as e:
           exceptions.append(e)
      if exceptions:
          ...  # Throw aggregated exception for potentially multiple download errors.